Doc/reference/expressions.rst

```diff
@@ -1649,7 +1649,32 @@ Assignment expressions
 .. productionlist::
    assignment_expression: [`identifier` ":="] `expression`
 
-.. TODO: BPO-39868
+Assignment expression is the new expression introduced in the Python3.8
+or later (sometimes also called as "named expression" or "walrus operator"),
+which aims to reduce the number of lines of code by allowing the
+assignment and return of value in the same expression. This makes codebase
+more readable and maintainable.
+
+The expression ``name := expr`` where **expr** is any valid Python expression
+other than an unparenthesize tuple and the **name** is an identifier.
+Here the variable **name** holds the assigned value of a given **expression**.
+
```

Comment:
I personally don't think this needs to include the history or motivation... just the behavior. We can reuse the above grammar for the explanation, too!

```suggestion
An assignment expression (sometimes also called a "named expression" or
"walrus") assigns an :token:`expression` to an :token:`identifier`, while also
returning the value of the :token:`expression`. 
```